src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```
@@ -417,24 +419,46 @@ public async Task<JObject> Resolve(InvocationExpressionSyntax method, Dictionary
+                            // explicitly passed arguments
+                            if (i < method.ArgumentList.Arguments.Count)
                             {
-                                if (!await commandParamsObjWriter.WriteConst(arg.Expression as LiteralExpressionSyntax, context.SdbAgent, token))
-                                    return null;
-                            }
-                            if (arg.Expression is IdentifierNameSyntax)
-                            {
-                                var argParm = arg.Expression as IdentifierNameSyntax;
-                                if (!await commandParamsObjWriter.WriteJsonValue(memberAccessValues[argParm.Identifier.Text], context.SdbAgent, token))
-                                    return null;
+                                var arg = method.ArgumentList.Arguments[i];
+                                if (arg.Expression is LiteralExpressionSyntax)
+                                {
+                                    if (!await commandParamsObjWriter.WriteConst(arg.Expression as LiteralExpressionSyntax, context.SdbAgent, token))
+                                        return null;
+                                }
+                                if (arg.Expression is IdentifierNameSyntax)
+                                {
+                                    var argParm = arg.Expression as IdentifierNameSyntax;
+                                    if (!await commandParamsObjWriter.WriteJsonValue(memberAccessValues[argParm.Identifier.Text], context.SdbAgent, token))
+                                        return null;
+                                }
                             }
+                            // optional arguments that were not overritten
+                            else if (!await commandParamsObjWriter.WriteConst(methodParamsInfo[i].TypeCode, methodParamsInfo[i].Value, context.SdbAgent, token))
+                                return null;
```

This should throw an exception, since failing to write the const here shouldn't be normal.